### PR TITLE
Style updates for the Applications section

### PR DESCRIPTION
**Pull request by ahardin-rh**

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@ahardin-rh We can chat on this tomorrow, but we had talked about not using "deployment config" as a general reference for the `DeploymentConfig` objects. And instead just using "deployment" as a general reference for both `Deployment` and `DeploymentConfig` objects (and just using the specific object when it's important to make the distinction).

See: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/term_glossary.adoc#deployment

Not sure how difficult these sections might be in this case, since there are a lot of specifics on differences between the two objects. Let's catch up on this tomorrow CC @adellape / @jboxman for thoughts, since this discussion/decision came from us all.

**Comment by codyhoag** [comment]
I'm going to hold off on peer-reviewing this until the Deployment/DeploymentConfig conversation is resolved. A large chunk of this PR is related to that decision :slightly_smiling_face: . From what I can tell, there will need to be a distinction made most of the time between `Deployment` and `DeploymentConfig`, since we're specifically explaining deployment strategies.

Let me know when this is settled. Thanks!

**Comment by ahardin-rh** [comment]
Reworked after discussion on the "deployment" style guidance.

File: modules/deployments-comparing-deploymentconfigs.adoc [hunk 1]
```diff
@@ -3,50 +3,33 @@
...
 
 [id="deployments-comparing-deploymentconfigs_{context}"]
-= Comparing Deployments and DeploymentConfigs
+= Comparing `Deployments` and `DeploymentConfig` objects
 
-Both Kubernetes Deployments and {product-title}-provided
-DeploymentConfigs are supported in {product-title}; however, it is
-recommended to use Deployments unless you need a specific feature or behavior
-provided by DeploymentConfigs.
+Both Kubernetes `Deployments` objects and {product-title}-provided `DeploymentConfig` objects are supported in {product-title}; however, it is recommended to use deployments unless you need a specific feature or behavior provided by `DeploymentConfig` objects.
```
[the remark this answers is not in the document]
Comment (codyhoag) [reply modules/deployments-comparing-deploymentconfigs.adoc:8]: I think mixing the proper object names and lower case references should be avoided within a single sentence (or thought). IMHO, consistency makes it clearer what we're talking about.

File: modules/deployments-comparing-deploymentconfigs.adoc [hunk 2]
```diff
@@ -62,44 +45,27 @@ Deployments do not yet support any lifecycle hooks.
...
 
 [discrete]
 ==== Rollover
 
-The deployment process for Deployments is driven by a controller
-loop, in contrast to DeploymentConfigs which use deployer pods for every
-new rollout. This means that a Deployment can have as many active
-ReplicaSets as possible, and eventually the deployment controller will scale
-down all old ReplicaSets and scale up the newest one.
+The deployment process for deployments is driven by a controller loop, in contrast to `DeploymentConfig` objects which use deployer pods for every new rollout. This means that a deployment can have as many active replica sets as possible, and eventually the deployment controller will scale down all old replica sets and scale up the newest one.
```
[the remark this answers is not in the document]
Comment (codyhoag) [reply modules/deployments-comparing-deploymentconfigs.adoc:56]: Same thought here about consistency when referring to object names vs. lower-case.

**Review by codyhoag — COMMENTED** [review]
Just a few suggestions, mostly related to consistency and keeping the plural `s` out of the monospace object names. Great work with the slew of *deployment* options :zany_face: .

**Pull request merged**